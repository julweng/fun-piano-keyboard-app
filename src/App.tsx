import React from "react"
import { notes } from "./Notes/noteType"
import Octave from "./components/Octave/Octave"
import "./App.css"

const App = () => {
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const audio = new Audio(`sounds/piano_${event.currentTarget.value}.mp3`)
    audio.play()
  }
  return (
    <div className="App">
      <header className="App-header">
        <h1>Mini Piano Keyboard</h1>
      </header>
      <Octave notes={notes} clickHandler={handleClick} />
    </div>
  )
}

export default App
