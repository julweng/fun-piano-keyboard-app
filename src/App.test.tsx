import React from "react"
import { render, screen } from "@testing-library/react"
import App from "./App"

jest.mock("./components/Octave/Octave.tsx", () =>
  jest.fn(() => <div>Octave</div>)
)

describe("App", () => {
  it("Should render App", () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  it("Should render Octave", () => {
    render(<App />)
    const Octave = screen.getByText("Octave")
    expect(Octave).toBeTruthy()
  })
})

