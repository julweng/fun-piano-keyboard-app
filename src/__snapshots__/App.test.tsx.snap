// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render App 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        Mini Piano Keyboard
      </h1>
    </header>
    <div>
      Octave
    </div>
  </div>
</div>
`;
